pkg/vere/disk.c

```diff
@@ -924,24 +923,22 @@ u3_disk_init(c3_c* pax_c, u3_disk_cb cb_u)
         return 0;
       }
     }
-  }
-
-  //  get the latest event number from the db
-  //
-  {
-    log_u->dun_d = 0;
-    c3_d fir_d;
 
-    if ( c3n == u3_lmdb_gulf(log_u->mdb_u, &fir_d, &log_u->dun_d) ) {
-      fprintf(stderr, "disk: failed to load latest event from database\r\n");
-      c3_free(log_u);
-      return 0;
+    if ( 0 == lat_d ) {
+      c3_d fir_d;
+      if ( c3n == u3_lmdb_gulf(log_u->mdb_u, &fir_d, &log_u->dun_d) ) {
+        log_u->dun_d = 0;
+      }
+      log_u->sen_d = log_u->dun_d;
+    } else {
+      log_u->dun_d = lat_d;
+      log_u->sen_d = log_u->dun_d;
```

Comment:
`sen_d` and `dun_d` are for "commits in progress" vs "commits completed" -- you should just add a new member to the `u3_disk` struct for "first event in epoch".

Reply:
Thanks, but I don't understand this yet. Why in the old code do we set `sen_d` to `dun_d`? Also, I think the incorrectness of my code here is why I can't boot rolled ships yet.

Reply:
Yeah, they're both event numbers. "commits in flight" would be `sen_d - dun_d`.